packages/repo/src/mst/mst.ts

```diff
@@ -669,6 +669,40 @@ export class MST implements DataStore {
     return leaves.length
   }
 
+  // Reachable tree traversal
+  // -------------------
+
+  // Walk reachable branches of tree & emit nodes, consumer can bail at any point by returning false
+  async *walkReachable(): AsyncIterable<NodeEntry> {
```

Comment:
This would be for an MST that is being read out of a blockstore that may not hall all relevant blocks.

The simplest case is a proof for a single record. Then, the only reachable nodes are the blocks in that proof & the only reachable leaf is the record itself

Reply:
Ahh, got it.  For some reason my brain imagined an MST with multiple components, i.e. having multiple roots and traversing nodes connected to just one of the roots.  Aaaaaanyways!